### PR TITLE
Country Feature Commit

### PR DESCRIPTION
**Pull request by arijit1111**
Pull request generated for Module Country

### CODE REVIEW

File: Modules/Country/Entities/City.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Modules\Country\Entities;
+
+use Illuminate\Database\Eloquent\Model;
+
+class City extends Model
+{
+
+    protected $fillable = [];
```
Comment (SonaKhachatryan1995) [inline Modules/Country/Entities/City.php]: Add fillables

File: Modules/Country/Entities/Country.php [hunk 2]
```diff
@@ -4,11 +4,9 @@
 
 use Illuminate\Database\Eloquent\Model;
 
-
 class Country extends Model
 {
-    protected $table = 'countries';
-    public $timestamps = true;
+
```
Comment (SonaKhachatryan1995) [inline Modules/Country/Entities/Country.php:9]: add fillables

File: Modules/Country/Entities/State.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Modules\Country\Entities;
+
+use Illuminate\Database\Eloquent\Model;
+
+class State extends Model
+{
+    protected $fillable = [];
```
Comment (SonaKhachatryan1995) [inline Modules/Country/Entities/State.php]: Add fillables

File: Modules/Country/Http/Controllers/CountryController.php [hunk 4]
```diff
@@ -4,85 +4,65 @@
 
 use Illuminate\Http\Request;
```
Comment (SonaKhachatryan1995) [inline Modules/Country/Http/Controllers/CountryController.php]: Remove this, not used

File: Modules/Country/Http/Requests/CityRequest.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules(): array
+    {
+        return [
+            'state_id' => 'required|integer'
```
Comment (SonaKhachatryan1995) [inline Modules/Country/Http/Requests/CityRequest.php]: Check if exists in state table

File: Modules/Country/Http/Requests/StateRequest.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, mixed>
+     */
+    public function rules(): array
+    {
+        return [
+            'country_id' => 'required|integer'
```
Comment (SonaKhachatryan1995) [inline Modules/Country/Http/Requests/StateRequest.php]: Check if exists in countries table

**Review by SonaKhachatryan1995 — CHANGES_REQUESTED** [review]
Remove use ```Illuminate\Database\Eloquent\Factory; ``` from ```CountryServiceProvider.php```

File: Modules/Country/Entities/City.php (hunk 1 above)
Comment (arijit1111) [reply Modules/Country/Entities/City.php]: Done

File: Modules/Country/Entities/Country.php (hunk 2 above)
Comment (arijit1111) [reply Modules/Country/Entities/Country.php:9]: Done

File: Modules/Country/Entities/State.php (hunk 3 above)
Comment (arijit1111) [reply Modules/Country/Entities/State.php]: Done

File: Modules/Country/Http/Controllers/CountryController.php (hunk 4 above)
Comment (arijit1111) [reply Modules/Country/Http/Controllers/CountryController.php]: Done

File: Modules/Country/Http/Requests/CityRequest.php (hunk 5 above)
Comment (arijit1111) [reply Modules/Country/Http/Requests/CityRequest.php]: Done

File: Modules/Country/Http/Requests/StateRequest.php (hunk 6 above)
Comment (arijit1111) [reply Modules/Country/Http/Requests/StateRequest.php]: Done

**Comment by arijit1111** [comment]
> 



> Remove use `Illuminate\Database\Eloquent\Factory; ` from `CountryServiceProvider.php`

Done

**Review by SonaKhachatryan1995 — APPROVED** [review]

**Pull request merged**